extensions/src/ACRE2Arma/common/pbo/search.cpp

```diff
@@ -128,35 +129,50 @@ namespace acre {
             return index_files(".*");
         }
 
-        bool search::index_files(const std::string &filter) {
+        using pbo_worker_result = std::tuple<std::string, std::unordered_map<std::string, std::string>>;
+        pbo_worker_result get_files_in_pbo(const std::string& pbo_file_path, const std::regex& txt_regex) {
+            std::unordered_map<std::string, std::string> pbo_files_map;
             std::ifstream pbo_stream;
+            pbo_stream.open(pbo_file_path, std::ios::binary | std::ios::in);
+            if (!pbo_stream.good()) {
+                return { pbo_file_path, pbo_files_map };
+            };
+            acre::pbo::archive _archive(pbo_stream);
+            for (acre::pbo::entry_p& entry : _archive.entries) {
+                if (entry->filename != "") {
+                    if (std::regex_match(entry->filename, txt_regex)) {
+                        std::string full_virtual_path = _archive.info->header["prefix"] + "\\" + entry->filename;
+                        std::transform(full_virtual_path.begin(), full_virtual_path.end(), full_virtual_path.begin(), ::tolower);
+                        pbo_files_map[full_virtual_path] = pbo_file_path;
+                    }
+                }
+            }
+            pbo_stream.close();
+            return { "", pbo_files_map };
+        }
+
+        bool search::index_files(const std::string& filter) {
             std::regex txt_regex(filter);
 
             if (_active_pbo_list.size() < 1)
                 return false;
 
+            std::vector<std::future<pbo_worker_result>> fWorkers;
             for (auto & pbo_file_path : _active_pbo_list) {
-                pbo_stream.open(pbo_file_path, std::ios::binary | std::ios::in);
-                if (!pbo_stream.good()) {
-                    LOG(ERROR) << "Cannot open file - " << pbo_file_path;
+                fWorkers.emplace_back(std::async(std::launch::async, &get_files_in_pbo, pbo_file_path, txt_regex));
+            }
+            for (auto& worker : fWorkers) {
```

Comment:
```suggestion
            for (const auto &worker : fWorkers) {
```

Reply:
`get()` won't work on const; the func can only be called once and leaves the future invalid

Reply:
oh true!